packages/client-react-streaming/src/DataTransportAbstraction/useTransportValue.tsx

```diff
@@ -17,19 +24,22 @@ export function useTransportValue<T>(value: T): T {
     throw new Error(
       "useTransportValue must be used within a streaming-specific ApolloProvider"
     );
-  const valueRef = dataTransport.useStaticValueRef(value);
+  const valueRef = dataTransport.useStaticValueRef<T | typeof CLEAN>(value);
 
-  const retVal = useSyncExternalStore(
+  const whichResult = useSyncExternalStore(
     () => () => {},
-    () => value,
-    () => valueRef.current
+    () => WhichResult.client,
+    () =>
+      valueRef.current === CLEAN
+        ? WhichResult.client
+        : equal(value, valueRef.current)
+          ? WhichResult.client
+          : WhichResult.server
   );
```

Comment:
Instead of flipping from `valueRef.current` to `value` here, this will flip from `1` to `0`, so a change from `value` in the future won't be seen as a change to the result of `uSES`.

Also, adding the `equal` check here means we can return the `value` here from the get-go if it's deeply equal with the transported value `valueRef.current` - which will further referentially stabilize values returned by this hook.